great_expectations/core/configuration.py

```diff
@@ -16,6 +17,9 @@ def __init__(self, id: Optional[str] = None, name: Optional[str] = None) -> None
         self.name = name
         super().__init__()
 
+    def __repr__(self) -> str:
+        return json.dumps(self.to_dict(), indent=2)
```

Comment:
Are we just using `json.dumps()` for the pretty formatting?
If so I would recommend using `pprint.pformat()` instead.
https://docs.python.org/3/library/pprint.html#pprint.pformat
```python
from pprint import pformat as pf

...

    def __repr__(self) -> str:
        return pf(self.to_dict(), indent=2)
```
Also, you can take advantage of nice formatting features like `depth` which will cause the formatter to only show structures up to `x` depth, and uses `...` for everything else.
```python
>>> pf({"a": {"1": 1}, "b": {"2": 2}, "c": [1, 2, 3]}, depth=1)
... {'a': {...}, 'b': {...}, 'c': [...]}
```

Reply:
love it!